apps/server/src/modules/sharing/api/test/sharing-import-room-token.api.spec.ts

```diff
@@ -0,0 +1,282 @@
+						parentId: room.id,
+						parentType: ShareTokenParentType.Room,
+						contextType: undefined,
+						contextId: undefined,
+					});
+
+					await em.persistAndFlush([
+						room,
+						roomMembership,
+						teacherAccount,
+						teacherUser,
+						userGroup,
+						roomOwnerRole,
+						shareToken,
+					]);
+					em.clear();
+
+					const loggedInClient = await testApiClient.login(teacherAccount);
+
+					return { loggedInClient, token: shareToken.token, room };
+				};
+				it('should return a 201 status', async () => {
+					const { loggedInClient, token } = await setup();
+					const response = await loggedInClient.post(`${token}/import`, { newName: 'NewName' });
+					expect(response.status).toBe(HttpStatus.CREATED);
+				});
+			});
+
+			describe('when the user does not have the required permissions', () => {
+				const setup = async () => {
+					const school = schoolEntityFactory.buildWithId();
+					const room = roomEntityFactory.buildWithId({ schoolId: school.id });
+					const { roomOwnerRole, roomViewerRole } = RoomRolesTestFactory.createRoomRoles();
+
+					const { teacherAccount, teacherUser } = UserAndAccountTestFactory.buildTeacher({ school });
+					const { studentAccount, studentUser } = UserAndAccountTestFactory.buildStudent({ school });
+
+					const userGroup = groupEntityFactory.buildWithId({
+						type: GroupEntityTypes.ROOM,
+						users: [{ role: roomOwnerRole, user: teacherUser }],
+					});
+					const roomMembership = roomMembershipEntityFactory.build({
+						roomId: room.id,
+						userGroupId: userGroup.id,
+						schoolId: school.id,
+					});
+
+					const userGroupStudent = groupEntityFactory.buildWithId({
+						type: GroupEntityTypes.ROOM,
+						users: [{ role: roomViewerRole, user: studentUser }],
+					});
+					const roomMembershipStudent = roomMembershipEntityFactory.build({
+						roomId: room.id,
+						userGroupId: userGroupStudent.id,
+						schoolId: school.id,
+					});
+
+					const shareToken = shareTokenFactory.build({
+						parentId: room.id,
+						parentType: ShareTokenParentType.Room,
+						contextType: undefined,
+						contextId: undefined,
+					});
+
+					await em.persistAndFlush([
+						room,
+						roomMembership,
+						roomMembershipStudent,
+						teacherAccount,
+						teacherUser,
+						studentAccount,
+						studentUser,
+						userGroup,
+						roomOwnerRole,
+						shareToken,
+					]);
+					em.clear();
+
+					const loggedInClientStudent = await testApiClient.login(studentAccount);
+
+					return { loggedInClientStudent, token: shareToken.token, room };
+				};
+				it('should return a 403 error', async () => {
+					const { loggedInClientStudent, token } = await setup();
+					const response = await loggedInClientStudent.post(`${token}/import`, { newName: 'NewName' });
+					expect(response.status).toBe(HttpStatus.UNAUTHORIZED);
+				});
+			});
+		});
+
+		describe('when the user is from a different school', () => {
```

Comment:
This test (Import to external school) fails.
Also see similar room-copy api test for user who is external.